src/Donors/Actions/CreateUserFromDonor.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Give\Donors\Actions;
+
+use Give\Donors\Models\Donor;
+
+/**
+ * @unreleased
+ */
+class CreateUserFromDonor
+{
+    public function __invoke(Donor $donor)
+    {
+        $userIdOrError = wp_insert_user(apply_filters(
+            'givewp_create_donor_new_user',
+            [
+                'user_login'      => $donor->email,
+                'user_pass'       => wp_generate_password(),
+                'user_email'      => $donor->email,
+                'first_name'      => $donor->firstName,
+                'last_name'       => $donor->lastName,
+                'role'            => give_get_option( 'donor_default_user_role', 'give_donor' ),
+            ],
+            $donor
+        ));
+
+        if(!is_wp_error($userIdOrError)) {
+            $donor->userId = $userIdOrError;
+        } else {
+            // How should we handle this?
+            throw new \Exception('Could not create user from donor');
```

Comment:
We should be catching/logging this during implementation, maybe we can add the WP_Error object message/code to our exception thrown to get more information.

Reply:
Resolved, see https://github.com/impress-org/givewp/pull/7103/commits/b83c387611ae7a4ac8ef88e5e61070681d972566 and https://github.com/impress-org/givewp/pull/7103/commits/eb5dc2d87ef625f2ab71eb01b9a6e54440967be5.